File: CoWorkHub/CoWorkHub.Model/Review.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CoWorkHub.Model
{
    public class Review
    {
        public int ReviewsId { get; set; }
        public int ReservationId { get; set; }
        public byte Rating { get; set; }
        public string Comment { get; set; } = null!;
        public DateTime CreatedAt { get; set; }
        public virtual Reservation Reservation { get; set; } = null!;
    }
}

File: CoWorkHub/UI/coworkhub_desktop/lib/models/review.dart
import 'package:coworkhub_desktop/models/reservation.dart';
import 'package:json_annotation/json_annotation.dart';

part 'review.g.dart';

@JsonSerializable()
class Review {
  final int reviewsId;
  final int reservationId;
  final int rating;
  final String comment;
  final DateTime createdAt;
  Reservation? reservation;

  Review({
    required this.reviewsId,
    required this.reservationId,
    required this.rating,
    required this.comment,
    required this.createdAt,
    this.reservation,
  });

  int? get spaceUnitId => reservation?.spaceUnitId;

  factory Review.fromJson(Map<String, dynamic> json) => _$ReviewFromJson(json);

  Map<String, dynamic> toJson() => _$ReviewToJson(this);
}
